components/PairingItem.js

```diff
@@ -1,40 +1,83 @@
+import { useState } from "react";
+import styled from "styled-components";
+import { nanoid } from "nanoid";
 import {
-  Pairing,
+  Card,
   Ingredients,
   Reason,
   ImageWrapper,
   StyledImage,
-  StyledContent2,
+  StyledContent,
   Flavors,
   CardFooter,
+  FavoriteButton,
+  Ingredient,
 } from "@/_styles";
 import { flavorColors } from "@/utils";
 import ingredientsData from "@/assets/ingredients.json";
+import CommentPopup from "@/components/CommentPopup";
+
+const PairingItem = ({ pairing, toggleFavoritePairing, isFavorite }) => {
+  const [showCommentPopup, setShowCommentPopup] = useState(false);
+  const [comments, setComments] = useState([]);
+  const [editingComment, setEditingComment] = useState(null);
 
-const PairingItem = ({ pairing }) => {
   const ingredients = pairing.ingredients.map((id) => {
     const ingredient = ingredientsData.find((ing) => ing._id === id);
     return ingredient;
   });
 
+  const handleCommentSubmit = (comment, commentId) => {
+    if (commentId) {
+      const updatedComments = comments.map((c) =>
+        c.id === commentId ? { ...c, text: comment } : c
+      );
+      setComments(updatedComments);
+    } else {
+      setComments([...comments, { id: nanoid(), text: comment }]);
+    }
+    setShowCommentPopup(false);
+    setEditingComment(null);
+  };
+
+  const handleEdit = (commentId) => {
+    const commentToEdit = comments.find((c) => c.id === commentId);
+    if (commentToEdit) {
+      setEditingComment(commentToEdit);
+      setShowCommentPopup(true);
+    }
+  };
+
+  const handleDelete = (commentId) => {
+    const updatedComments = comments.filter((c) => c.id !== commentId);
+    setComments(updatedComments);
+  };
```

Comment:
```suggestion
  const handleDelete = (commentId) => {
    const updatedComments = comments.filter((c) => c.id !== commentId);
    setComments(updatedComments);
    setShowCommentPopup(false);
  };
```

Reply:
Ideally, when the user clicks on Delete button, the comments window should close.

Reply:
Thank you Sabeen for your suggestion! 
I updated the code to close the comment window when the delete button is clicked.